skbio/io/registry.py

```diff
@@ -514,24 +533,44 @@ def _read_gen(self, file, fmt, into, verify, kwargs):
         # kwargs should still retain the contents of io_kwargs because the
         # actual reader will also need them.
         with _resolve_file(file, **io_kwargs) as (file, _, _):
-            reader, kwargs = self._init_reader(
+            seekable = True
+            try:
+                file.tell()
+            except io.UnsupportedOperation:
+                seekable = False
+            if not seekable and (fmt is not None or not fmt.support_non_seekable):
```

Comment:
Here is a problem. This `fmt` is a string. It doesn't have properties like `support_non_seekable`. Similar above.